ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/PointSchedule.hs

```diff
@@ -151,34 +155,40 @@ data Tick =
 instance Condense Tick where
   condense Tick {active} = condense active
 
-data PointSchedule =
-  PointSchedule { ticks :: [Tick], frags :: Peers TestFrag }
+newtype PointSchedule =
+  PointSchedule {ticks :: NonEmpty Tick}
```

Comment:
Initially, we also required each individual peer to have nonempty schedules, but we weren't sure that it was required (also it was painful to work with).
Is there a reason we should choose either of those? An empty fork doesn't make any sense, but the chain generator probably prevents that anyway.

Reply:
The change would be to disallow peer fragments that only consist of the genesis point. Right now that's accepted. The benefit might be that the point schedule generator would get nonempty chains as input, which could make some of the types easier, but maybe some of them more convoluted. I'm unsure what would be better.